pkg/controllermanager/controller/project/project_stale_control.go

```diff
@@ -0,0 +1,343 @@
+	// exit exceptionally at any point provided they wish the update to be re-run at a later point in time.
+	ReconcileStaleProject(project *gardencorev1beta1.Project) error
+}
+
+// NewDefaultStaleControl returns a new instance of the default implementation StaleControlInterface that
+// implements the documented semantics for Projects. updater is the UpdaterInterface used
+// to update the status of Projects. You should use an instance returned from NewDefaultStaleControl() for any
+// scenario other than testing.
+func NewDefaultStaleControl(k8sGardenClient kubernetes.Interface,
+	k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory,
+	config *config.ControllerManagerConfiguration,
+	recorder record.EventRecorder,
+	shootLister gardencorelisters.ShootLister,
+	plantLister gardencorelisters.PlantLister,
+	backupEntryLister gardencorelisters.BackupEntryLister,
+	secretBindingLister gardencorelisters.SecretBindingLister,
+	quotaLister gardencorelisters.QuotaLister,
+	secretLister kubecorev1listers.SecretLister,
+) StaleControlInterface {
+	return &defaultStaleControl{k8sGardenClient,
+		k8sGardenCoreInformers,
+		config,
+		recorder,
+		shootLister,
+		plantLister,
+		backupEntryLister,
+		secretBindingLister,
+		quotaLister,
+		secretLister,
+	}
+}
+
+type defaultStaleControl struct {
+	k8sGardenClient        kubernetes.Interface
+	k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory
+	config                 *config.ControllerManagerConfiguration
+	recorder               record.EventRecorder
+	shootLister            gardencorelisters.ShootLister
+	plantLister            gardencorelisters.PlantLister
+	backupEntryLister      gardencorelisters.BackupEntryLister
+	secretBindingLister    gardencorelisters.SecretBindingLister
+	quotaLister            gardencorelisters.QuotaLister
+	secretLister           kubecorev1listers.SecretLister
+}
+
+// NowFunc is a function that returns the current time.
+var NowFunc = metav1.Now
+
+func (c *defaultStaleControl) ReconcileStaleProject(obj *gardencorev1beta1.Project) error {
+	if obj.Spec.Namespace == nil {
+		return nil
+	}
+
+	var (
+		ctx           = context.TODO()
+		project       = obj.DeepCopy()
+		projectLogger = newProjectLogger(project)
+	)
+
+	projectLogger.Infof("[STALE PROJECT RECONCILE]")
+
+	// Skip projects that are not older than the configured minimum lifetime in days. This allows having Projects for a
+	// certain period of time until they are checked whether they got stale.
+	if project.CreationTimestamp.UTC().Add(time.Hour * 24 * time.Duration(*c.config.Controllers.Project.MinimumLifetimeDays)).After(NowFunc().UTC()) {
+		projectLogger.Infof("[STALE PROJECT RECONCILE] Project is not older than the configured minimum %d days lifetime (%v), considering it 'not stale'", *c.config.Controllers.Project.MinimumLifetimeDays, project.CreationTimestamp.UTC())
+		return c.markProjectAsNotStale(ctx, project)
```

Comment:
Thanks for the review. My thoughts were that we don't want to "frighten" people that create new projects with an immediate "stale" notification. But this whole thing is configurable, so if someone doesn't want this behaviour then simply `minimumLifetimeDays` can be set to `0`.

Reply:
Sure, I get the motivation and it's good to have it. My comment was more logic wise because I thought a "new" project doesn't have stale timestamps at all, does it? But you probably mark it as not stale anytime in case the `MinimumLifetimeDays` in the configuration has been prolonged, right? 👀

Reply:
Ah, I see. Yes, exactly! It was created as part of @tim-ebert's thoughts (https://github.com/gardener/gardener/issues/1997#issuecomment-614436902) to support changing these fields dynamically.